File: solutions/_14_longest_common_prefix/solution.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ""

        for i in range(len(strs[0])):
            char = strs[0][i]
            for s in strs[1:]:
                if i >= len(s) or s[i] != char:
                    return res
            res += char

        return res


# https://leetcode.com/problems/longest-common-prefix/

File: solutions/_14_longest_common_prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  let res = "";
  for (let i = 0; i < strs[0].length; i++) {
    const char = strs[0][i];
    if (strs.every((str) => str[i] === char)) {
      res += char;
    } else break;
  }
  return res;
};

module.exports = { longestCommonPrefix };

/*
https://leetcode.com/problems/longest-common-prefix/
*/
